src/OrbitGl/TimerTrack.cpp

```diff
@@ -101,42 +125,112 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
         if (!TimerFilter(timer_info)) continue;
-        uint64_t function_id = timer_info.function_id();
 
         UpdateDepth(timer_info.depth() + 1);
         double start_us = time_graph_->GetUsFromTick(timer_info.start());
+        double start_or_prev_end_us = start_us;
         double end_us = time_graph_->GetUsFromTick(timer_info.end());
-        double elapsed_us = end_us - start_us;
-        double normalized_start = start_us * inv_time_window;
-        double normalized_length = elapsed_us * inv_time_window;
-        float world_timer_width = static_cast<float>(normalized_length * world_width);
-        float world_timer_x = static_cast<float>(world_start_x + normalized_start * world_width);
+        double end_or_next_start_us = end_us;
+
         float world_timer_y = GetYFromTimer(timer_info);
+        float box_height = GetTextBoxHeight(timer_info);
+
+        // Check if the previous timer overlaps with the current one, and if so draw the overlap
+        // as triangles rather than as overlapping rectangles.
+        if (k > 0) {
+          const TimerInfo& prev_timer_info = block[k - 1].GetTimerInfo();
+          // We also compare the type, as for the Gpu timers, timers of different type but same
+          // depth are drawn below each other (and thus do not overlap).
+          if (prev_timer_info.end() > timer_info.start() &&
+              prev_timer_info.type() == timer_info.type()) {
+            start_or_prev_end_us = time_graph_->GetUsFromTick(prev_timer_info.end());
+          }
+        }
+
+        // Check if the next timer overlaps with the current one, and if so draw the overlap
+        // as triangles rather than as overlapping rectangles.
+        if (k + 1 < block.size()) {
+          const TimerInfo& next_timer_info = block[k + 1].GetTimerInfo();
+          // We also compare the type, as for the Gpu timers, timers of different type but same
+          // depth are drawn below each other (and thus do not overlap).
+          if (timer_info.end() > next_timer_info.start() &&
+              next_timer_info.type() == timer_info.type()) {
+            end_or_next_start_us = time_graph_->GetUsFromTick(next_timer_info.start());
+          }
+        }
+
+        double elapsed_us = end_us - start_us;
+
+        // Draw the Timer's text if it is not collapsed.
+        if (!is_collapsed) {
+          // For overlaps, let us make the textbox just a bit wider:
+          double left_overlap_width_us = start_or_prev_end_us - start_us;
+          double text_x_start_us = start_or_prev_end_us - (.25 * left_overlap_width_us);
+          double right_overlap_width_us = end_us - end_or_next_start_us;
+          double text_x_end_us = end_or_next_start_us + (.25 * right_overlap_width_us);
+          WorldXInfo world_x_info =
+              ToWorldX(text_x_start_us, text_x_end_us, inv_time_window, world_start_x, world_width);
+
+          if (world_x_info.is_visible_width) {
```

Comment:
Maybe I'm looking at the wrong thing, but it seems `is_visible_width` is never set in `ToWorldX`.

Reply:
Ah thanks, missed that code branch when removing. Should be fixed now.

Reply:
I don't see the change, yet. Can you check again?

Reply:
Well commit and then push is sometimes hard.... sorry for that. should be there now.